src/index.js

```diff
@@ -9,15 +10,15 @@ function statusCodeError (code) {
   return error;
 }
 
-module.exports = function retryDownload (retries, ...args) {
-  const filename = tempy.file();
+module.exports = function retryDownload (uri, retries = 3, requestOpts = {}) {
+  const filename = tempy.file({ extension: path.extname(uri) });
```

Comment:
would it be worth moving file creation into the response block once we know we have something to write? rather than creating a stream that won't be used on request failure?

Reply:
@aeh Fair call!